libs/ui-toolkit/src/components/toast/use-toasts.ts

```diff
@@ -1,11 +1,23 @@
 import { create } from 'zustand';
+import { immer } from 'zustand/middleware/immer';
 import type { Toast } from './toast';
+import hash from 'stable-hash';
+import omit from 'lodash/omit';
 
-type ToastsStore = {
-  /**
-   * A list of active toasts
-   */
-  toasts: Toast[];
+export type Toasts = Record<string, Toast>;
+
+const isEqual = (a: Toast, b: Toast) => {
+  const h1 = hash(omit(a, 'onClose'));
```

Comment:
suggestion (non blocking): Maybe use lodash/isEqual instead of introducing a new dependency. This is already in use in the project and will do a deep comparison. I think it should work here

Reply:
Oops, I was just testing over few things here with a `stable-hash`, shouldn't have left it here. Removed.